### ISSUE TITLE
IstioOperator API: Labels for the gateway don't work

### ISSUE THREAD

**Issue by iusergii** [istio/istio#22389]
**Bug description**
The IstioOperator API (install.istio.io/v1alpha1) GatewaySpec **label** configuration doesn't work. 

[ X] Configuration Infrastructure
[ ] Docs
[ X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
Deployment for istio-ingressgateway should have configured labels.

**Steps to reproduce the bug**
```yaml apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  components:
    ingressGateways:
    - name: istio-ingressgateway
      enabled: true
      label:
        ProductComponents: Istio
```
```bash
istioctl manifest apply -f istio-ingressgateway.yaml
```
Deployment/pod - doesn't have label `ProductComponents: Istio`

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
```bash 
client version: 1.5.0
control plane version: 1.5.0
data plane version: 1.5.0 (4 proxies)
```

**Comment by iusergii** [comment]
Probably caused by same issue: https://github.com/istio/istio/issues/21103

**Comment by ostromart** [comment]
It doesn't look like the plumbing has been implemented yet for IstioOperator for labels. As a workaround you can still use the Helm passthrough API for this (values.gateways.istio-ingressgateway.labels). Fix should be in the next patch version.

**Comment by SeanWallace** [comment]
The only problem with using `values.gateways.istio-ingressgateway.labels` is those apply to _all_ gateways. This makes it impossible to create a Gateway object with a selector that only matches a single gateway, for example.

Given the scope of the label is inside an individual gateway my expectation was that it would add those labels to that gateway only. Perhaps this was an unfair assumption.

**Comment by ostromart** [comment]
Yes, that's quite right, the values workaround will apply to all gateways while the field in components will be per gateway. We'll get the fix out asap.

**Comment by dgn** [comment]
@ostromart I noticed that we handle podAnnotations in the k8s overlay but labels are in the gatewaySpec - shouldn't they be moved as well?

**Comment by zombor** [comment]
Hi, this is blocking our upgrade to 1.5. Will this be in 1.5.2? We were hoping multi-gateway functionality would be all cleaned up and properly working in 1.5.1 since there were multiple problems in the 1.5.0 release.

**Comment by lmarszal** [comment]
I run into the same problem too. Quick "hotfix" doesn't seem to be difficult - not sure if it's the best place to put that logic https://github.com/istio/istio/pull/22840

If you want to test it out, I've compiled 1.5 with this patch https://github.com/lmarszal/istio/releases/tag/1.5.1-patch-gwlabel

**Comment by l8huang** [comment]
Hi @lmarszal, I'm facing same problem, thanks for your fix. Looks like your PR is still blocked by `cla/google — Need a CLA for one or more commit authors`, could you please take a look?

**Comment by lmarszal** [comment]
@l8huang ostromart implemented it yesterday too (don't know if he got inspired by my draft or not) https://github.com/istio/istio/pull/22981

@ostromart Can we expect backport of this feature to 1.5? It would be super helpful

**Comment by ostromart** [comment]
Hi @lmarszal I chatted about it with the 1.5 release manager and it will be included for the next 1.5 patch. Thanks for doing the work on that, sorry I didn't see your earlier work before.
